Sources/XCTest/Public/XCTestMain.swift

```diff
@@ -59,10 +59,10 @@
 ///
 /// - Parameter testCases: An array of test cases run, each produced by a call to the `testCase` function
 /// - seealso: `testCase`
-public func XCTMain(_ testCases: [XCTestCaseEntry]) -> Never {
+public func XCTMain(_ testCases: [XCTestCaseEntry], arguments: [String] = CommandLine.arguments) -> Never {
     let testBundle = Bundle.main
 
-    let executionMode = ArgumentParser().executionMode
+    let executionMode = ArgumentParser(arguments: arguments).executionMode
```

Comment:
The `ArgumentParser` initializer has a default value for its `arguments:` parameter of `CommandLine.arguments`. Can we remove that default parameter value now, to enforce that the caller must specify it? That seems best now, given that the caller is the place where this default value is defined.